src/services/housing.js

```diff
@@ -21,49 +21,76 @@ import './user'; // Needed for typedef of StudentProfileInfo
 
 /**
  * @global
- * @typedef ApartmentApplicant
+ * @typedef ApartmentApplicant Applicant info used by the student application page
  * @property {StudentProfileInfo} Profile The StudentProfileInfo object representing this applicant
  * @property {String} OffCampusProgram The name of department of this applicant's off-campus program, or 'None'
  */
 
 /**
  * @global
- * @typedef ApplicantPartialInfo
+ * @typedef FullApplicantInfo Applicant info used by the staff menu
+ * @property {Number} ApplicationID Application ID number of this application
  * @property {String} Username The username of this applicant
+ * @property {Number} Age The age of the student (in years) (only visible to housing admin)
  * @property {String} OffCampusProgram The name of department of this applicant's off-campus program, or 'None'
+ * @property {String} Probation Indicates whether the student has a disiplinary probation (visble only to housing admin)
+ * @property {Number} Points The number of application points for this student (only visible to housing admin)
  */
 
 /**
  * Note: Properties 'HallRank' and 'HallName' must be capitalized to match the backend
  * @global
  * @typedef ApartmentChoice
+ * @property {Number} [ApplicationID] Application ID number of this application
```

Comment:
Why is this in brackets?

Reply:
It's means it is optional,

Almost like how we can do `string?` in the C# view models to say that it is valid for that property to be entirely null

In the JS, it doesn't actually enforce those typing rules, so it is mostly just there for devs and documentation

It is optional in this case because the frontend doesn't need to store that value there when it does the save stuff, but when it is used to load an application it will expect to see the value filled in by the backend,

So it is masked as optional, because the custom object is equally valid with or without that property

Reply:
Okay, got it,